Source/WebCore/PAL/pal/PALSwift/CryptoKitShim.swift

```diff
@@ -44,153 +48,76 @@ public enum ErrorCodes: Int {
 private class Utils {
     static let zeroArray = [UInt8](repeating: 0, count: 0)
 }
+public struct AesGcmRV {
+    public var cipherText: OptionalVectorUInt8 = OptionalVectorUInt8()
+    public var errorCode: ErrorCodes = .success
+}
 
 public class AesGcm {
     public static func encrypt(
-        _ key: UnsafePointer<UInt8>,
-        keySize: UInt,
-        iv: UnsafePointer<UInt8>,
-        ivSize: UInt,
-        additionalData: UnsafePointer<UInt8>,
-        additionalDataSize: UInt,
-        plainText: UnsafePointer<UInt8>,
-        plainTextSize: UInt,
-        cipherText: UnsafeMutablePointer<UInt8>
-    ) -> ErrorCodes {
+        key: SpanConstUInt8, iv: SpanConstUInt8, ad: SpanConstUInt8, message: SpanConstUInt8,
+        desiredTagLengthInBytes: Int
+    ) -> AesGcmRV {
+        var rv = AesGcmRV()
         do {
-            if keySize > Int.max
-                || ivSize > Int.max
-                || plainTextSize > Int.max
-                || additionalDataSize > Int.max
-            {
-                return .tooBigArguments
-            }
-            if ivSize == 0 {
-                return .invalidArgument
-            }
-            let nonce = try AES.GCM.Nonce(
-                data: UnsafeBufferPointer(start: iv, count: Int(ivSize)))
-            var message = UnsafeBufferPointer(start: plainText, count: Int(plainTextSize))
-            if plainTextSize == 0 {
-                Utils.zeroArray.withUnsafeBufferPointer { ptr in
-                    message = ptr
-                }
+            if iv.size() == 0 {
+                rv.errorCode = .invalidArgument
+                return rv
             }
-            let symKey = SymmetricKey(
-                data: UnsafeBufferPointer(start: key, count: Int(keySize)))
-            let sealedBox: AES.GCM.SealedBox
-            if additionalDataSize > 0 {
-                sealedBox = try AES.GCM.seal(
-                    message,
-                    using: symKey,
-                    nonce: nonce,
-                    authenticating: UnsafeBufferPointer(
-                        start: additionalData, count: Int(additionalDataSize))
-                )
-            } else {
-                sealedBox = try AES.GCM.seal(
-                    message,
-                    using: symKey,
-                    nonce: nonce
-                )
+            let sealedBox: AES.GCM.SealedBox = try AES.GCM.seal(message, key: key, iv: iv, ad: ad)
+            var ciphertext: VectorUInt8 = Cpp.createVectorUInt8FromSpan(
```

Comment:
I think the way to do this is to have an unsafe method on ContiguousBytes that invokes the Vector constructor. I believe that sealedBox.ciphertext conforms to ContiguousBytes. Then you can do sealedBox.ciphertext.copyToVector() or copyToVector(sealedBox.ciphertext) or similar.